src/Utilities/VersionCheck.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import axios from 'axios';
+import logger from './Logger';
+import { version as packageVersion } from '../../package.json';
+import moment from 'moment';
+import _ from 'lodash';
+
+const VERSION_CHECK_DATE = 'VERSION_CHECK_DATE'; 
+const getVersionFile = async () => {
+    return axios.get<string>(`/version.txt?cache_bust=${new Date().getTime()}`);
+};
+
+const versionRefresh = async () => {
+    // To test in development make a version.txt in the public folder with a value that matches or does not match package.json (i.e. "1.5.5")
+    // Then comment out this check
+    // version.txt in production is generated in post build
+    if (process.env.NODE_ENV !== 'production') {
+        logger.debug('Version Check: No cache buster in development');
+        return;
+    }
+    try {
+        const versionAxiosResponse = await getVersionFile();
+        const versionResponse = versionAxiosResponse.data;
+        if (packageVersion.trim() !== versionResponse.trim()) {
+            logger.warn(`Version Check: Version mismatch packageVersion ${packageVersion} !== versionResponse ${versionResponse}`);
+            const versionCheckDate = localStorage.getItem(VERSION_CHECK_DATE);
+            let shouldReload = false;
+            if (_.isNil(versionCheckDate)) {
+                shouldReload = true;
+                logger.debug('Version Check: Time to try');
+            } else if (moment().isAfter(moment(versionCheckDate))) {
+                shouldReload = true;
+                logger.debug(`Version Check: Time to try again ${versionCheckDate}`);
+            } else {
+                shouldReload = false;
+                logger.error(`Version Check: Version mismatch packageVersion ${packageVersion} !== versionResponse ${versionResponse}. Too soon to try again ${versionCheckDate}`);
```

Comment:
This error seems like it would bloat the logs quickly.

Reply:
Nope, this can happen at most once per day per user (unless they clear all of their local storage but are still loading the cached page)
And in theory this can only happen after around realeases
This does not happen on the initial failure either, it is specifically after a hard reload failed

Reply:
So this is an error that can happen? Not TSNH? 🤔 

I don't think it's an error if a user clears their cache, that seems like a warning at most, but I'll approve it if you disagree

Reply:
Yeah I agree I think it should probably be a warning

That being said I would like this to be sent up to the server at least for now to gauge the solution (based on the jira comment)

Maybe it also makes sense to push warnings up to the server 🤷‍♂️ 

I would leave it for now until we figure out the above comment